### PR TITLE
Improve and update the MRTCore apps

### PR DESCRIPTION
**Pull request by rohanp-msft**
This change does the following:
- For all the samples, set/correct the platform min version to 17763.
- For the packaged samples, update to NuGet v8.0.1
- For all the samples, ensure they build and run on x86 and x64 (Debug and Release). For arm64 (Debug and Release), just building is verified as part of this change.
- For the .NET samples, don't override the .NET 5 targeting packages anymore: the version we get with the latest .NET 5 is newer

**Verified**
For all these sample apps, I verified that they build and run for x86 and x64, for both Debug and Release. For arm64, I just verified that they build for Debug and Release. Verifying that they run on arm64 will be done separately.
- WinUI C++ sample app
- WinUI C# sample app
- Console unpackaged sample app
- WinForms unpackaged sample app
- WPF

### CODE REVIEW

File: Samples/ResourceManagement/cpp-console-unpackaged/packages.config [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-  <package id="Microsoft.ProjectReunion" version="0.8.0-preview" targetFramework="native" />
-  <package id="Microsoft.ProjectReunion.DWrite" version="0.8.0-preview" targetFramework="native" />
-  <package id="Microsoft.ProjectReunion.Foundation" version="0.8.0-preview" targetFramework="native" />
-  <package id="Microsoft.ProjectReunion.InteractiveExperiences" version="0.8.0-preview" targetFramework="native" />
-  <package id="Microsoft.ProjectReunion.WinUI" version="0.8.0-preview" targetFramework="native" />
-  <package id="Microsoft.Windows.CppWinRT" version="2.0.210505.3" targetFramework="native" />
-  <package id="Microsoft.Windows.ImplementationLibrary" version="1.0.210204.1" targetFramework="native" />
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Microsoft.ProjectReunion" version="0.8.0-preview" targetFramework="native" />
```
Comment (angelazhangmsft) [inline Samples/ResourceManagement/cpp-console-unpackaged/packages.config:3]: Should these be 0.8.1?

Comment (angelazhangmsft) [reply Samples/ResourceManagement/cpp-console-unpackaged/packages.config:3]: nevermind, I see this is the unpackaged app

**Comment by angelazhangmsft** [comment]
LGTM, the only thing I would suggest is adding a note to the first section of the readme that the unpackaged samples are an experimental feature. Maybe:

**Note**: The unpackaged sample apps reference an experimental version of the Windows App SDK. Unpackaged app deployment is not currently supported for use by apps in production environments.

**Comment by rohanp-msft** [comment]
> LGTM, the only thing I would suggest is adding a note to the first section of the readme that the unpackaged samples are an experimental feature. Maybe:
> 
> **Note**: The unpackaged sample apps reference an experimental version of the Windows App SDK. Unpackaged app deployment is not currently supported for use by apps in production environments.

Done! But I added a link as well. If you think it's better left out, please lmk.

File: Samples/ResourceManagement/README.md [hunk 2]
```diff
@@ -8,7 +8,9 @@ products: windows
 
 # Load resources using MRT Core
 
-MRT Core is the modern resource loading library used in WinUI 3 and the Windows App SDK. These samples demonstrate how to use MRT Core to load resources from multiple resource files and how to select which version of a specific named resource to load. 
+MRT Core is the modern resource loading library used in WinUI 3 and the Windows App SDK. These samples demonstrate how to use MRT Core to load resources from multiple resource files and how to select which version of a specific named resource to load.
+
+**Note**: The unpackaged sample apps reference an experimental version of the Windows App SDK ([link](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/experimental-channel)). Unpackaged app deployment is not currently supported for use by apps in production environments.
```
Comment (angelazhangmsft) [inline Samples/ResourceManagement/README.md:13]:
```suggestion
**Note**: The unpackaged sample apps reference an [experimental version](https://docs.microsoft.com/windows/apps/windows-app-sdk/experimental-channel) of the Windows App SDK. Unpackaged app deployment is not currently supported for use by apps in production environments.
```

Comment (rohanp-msft) [reply Samples/ResourceManagement/README.md]: Done

**Review by angelazhangmsft — APPROVED** [review]

**Review by axelandrejs — APPROVED** [review]

**Pull request merged**